Sourcery/Parsing/Composer.swift

```diff
@@ -91,13 +91,74 @@ struct Composer {
         return unique.values.sorted { $0.name < $1.name }
     }
 
+    private func resolveGenerics(typeName: TypeName, unique: [String: Type]) {
+        let genericSplit = typeName.unwrappedTypeName.characters.split(separator: "<", maxSplits: 1)
+
+        guard genericSplit.count == 2 else {
+            return
+        }
+
+        let typeCharacters = genericSplit[0]
+        var genericCharacters = genericSplit[1]
+
+        guard genericCharacters.count > 1, genericCharacters.removeLast() == ">" else {
+            return
+        }
+
+        var genericTypeStrings = [String]()
+
+        var subGenericsCounter = 0
+        var characterBuffer = [Character]()
+
+        for character in genericCharacters {
```

Comment:
Very hard to follow this method and I'm not sure it's doing the right thing. There are already few extension for `String` that will help and I think the whole method can be simplified to something like this (also applying my previous comment about using `TypeParameter` type):

```swift
    private func parseGenericType(typeName: TypeName) {
        let genericComponents = typeName.unwrappedTypeName
            .characters.split(separator: "<", maxSplits: 1)
            .map({ String($0).stripped() })

        guard genericComponents.count == 2 else {
            return typeName
        }

        let name = genericComponents[0]
        let typeParametersString = genericComponents[1].dropLast()

        typeName.generic = GenericType(name: name, typeParameters: parseGenericTypeParameters(typeParametersString))
    }
    
    private func parseGenericTypeParameters(typeParametersString: String) -> [TypeParameter] {
        let typeParametersList = typeParametersString
            .commaSeparated()
            .map({
                let typeName = TypeName($0.stripped())
                resolveGenerics(typeName: typeName)
                return TypeParameter(typeName: typeName)
        })
    }
```

Resolving type parameter type should happen outside this method, similar to resolving tuple elements types. This way parsing generic type matches closer to other similar methods.